packages/frontend/src/components/chat.tsx

```diff
@@ -45,37 +52,108 @@ function Message({ msg, from, peerId }: MessageProps) {
 export default function ChatContainer() {
   const { libp2p } = useLibp2pContext()
   const { messageHistory, setMessageHistory } = useChatContext();
+  const { files, setFiles } = useFileChatContext();
   const [input, setInput] = useState<string>('')
+  const fileRef = useRef<HTMLInputElement>(null);
 
   // Effect hook to subscribe to pubsub events and update the message state hook
   useEffect(() => {
-    const messageCB = (evt: CustomEvent<Message>) => {
+    const messageCB = async (evt: CustomEvent<Message>) => {
       console.log('gossipsub console log', evt.detail)
       // FIXME: Why does 'from' not exist on type 'Message'?
       const { topic, data } = evt.detail
+
+      switch (topic) {
+        case CHAT_TOPIC: {
+          chatMessageCB(evt, topic, data)
+          break
+        }
+        case CHAT_FILE_TOPIC: {
+          chatFileMessageCB(evt, topic, data)
+          break
+        }
+        default: {
+          console.log(`Unexpected gossipsub topic: ${topic}`)
+          break
+        }
+      }
+    }
+
+    const chatMessageCB = (evt: CustomEvent<Message>, topic: string, data: Uint8Array) => {
       const msg = new TextDecoder().decode(data)
       console.log(`${topic}: ${msg}`)
 
       // Append signed messages, otherwise discard
       if (evt.detail.type === 'signed') {
-        setMessageHistory([...messageHistory, { msg, from: 'other', peerId: evt.detail.from.toString() }])
+        setMessageHistory([...messageHistory, { msg, fileObjectUrl: undefined, from: 'other', peerId: evt.detail.from.toString() }])
+      }
+    }
+
+    const chatFileMessageCB = async (evt: CustomEvent<Message>, topic: string, data: Uint8Array) => {
+      const fileId = new TextDecoder().decode(data)
+      console.log(`${topic}: ${fileId}`)
```

Comment:
Please remove

Reply:
Oh, I forgot to remove these debug logs. Thank you. I just removed all: 0f0e4b8